### DISCUSSION TITLE
Interpretation of one-shot NAS output

### DISCUSSION THREAD

**Discussion by sayakpaul** [microsoft/nni#3929] [Q&A]
I am referring to this example: [`notebooks/Retiarii_example_one-shot_NAS.ipynb`](https://github.com/microsoft/nni/blob/master/examples/notebooks/Retiarii_example_one-shot_NAS.ipynb). The workflow is clear to me and also thanks to the brief notes included in the notebook that clarifies the strategy too. 

Here's what we get as a final output:

```
Final architecture: {'_mutation_1': 1, '_mutation_2': 1, '_mutation_3': [1]}
```

How to interpret this?

**Comment by ultmaster** [comment]
This is the mapping from choice label to choice index. In the example, all the labels along with index are autogenerated. In case you want to use your own, you can write

```python
self.choice = nn.LayerChoice({'conv1': ..., 'conv2': ...}, label='my_layer_1')
```

**Reply by sayakpaul** [reply]
Okay. It's still a bit vague. I think a more elaborate explanation on how to interpret the final output and create the final model in the tutorial would be more helpful.
